### PR TITLE
added pagination to `file_list`

### LINKED ISSUE #1650: Add pagination to file_list RPC command
`file_list` results should be paginated, just like other `*_list` RPC commands. 

Might be blocked by https://github.com/lbryio/lbry/issues/1228.

### PR DESCRIPTION
**Pull request by osilkin98**
fixes #1650

### CODE REVIEW

File: lbry/lbry/extras/daemon/Daemon.py [hunk 1]
```diff
@@ -1544,15 +1545,32 @@ def jsonrpc_file_list(self, sort=None, reverse=False, comparison=None, **kwargs)
...
+            offset = page_size * (page-1)
+            return {
+                'total_items': total_items,
+                'total_pages': int((total_items + (page_size-1)) / page_size),
+                'page': page,
+                'page_size': page_size,
+                'items': file_list[offset:offset+page_size]
+            }
+
+        return file_list
```
Comment (eukreign) [inline lbry/lbry/extras/daemon/Daemon.py]: The method should return the same return structure (either `Paginated` or `List` but not both).

Comment (osilkin98) [reply lbry/lbry/extras/daemon/Daemon.py]: I fixed by defaulting `page` and `page_size` to `1` and `10` respectively.

**Comment by osilkin98** [comment]
I cannot add it myself but this PR needs the `backwards-incompatible` tag on it.

**Comment by tzarebczan** [comment]
FYI: We have some inconsistencies for pagination across other apis - i.e. transaction/support list will show all transactions, and not paginate by default, but claim search does paginate.

**Comment by tzarebczan** [comment]
Lex also mentioned that we aren't using the same pagination process as other APIs - is that possible here or it requires deeper changes that @shyba was looking into?

**Comment by osilkin98** [comment]
@tzarebczan Should it use pagination only when `page` and `page_size` are passed in, or just either one (with the other defaulted)? 

> Lex also mentioned that we aren't using the same pagination process as other APIs - is that possible here or it requires deeper changes that @shyba was looking into?

This will have to be done in a separate PR.

**Comment by tzarebczan** [comment]
I'd say only use it when it's passed - this would be backward compatible and we can make larger breaking changes when we reformat the entire output to use claim data.

**Comment by tzarebczan** [comment]
Not sure if adding pagination before doing that / doing it the right way is useful though...

**Comment by osilkin98** [comment]
@tzarebczan Good call, existing code won't need to be refactored as done in later commits.

**Comment by jackrobison** [comment]
I don't think we should introduce multiple response formats that are contingent on the args provided to an api call - this will only add handling code for clients of the api for the different cases and push back the BC break a little further. If pagination is to become the default, then it should always return the dict of paginated results. Updating to handle paginated results should be fairly simple for api clients - this smaller BC break shouldn't be blocked by future changes to claims within file_list.

**Comment by lyoshenka** [comment]
Feels like the solution is to pretend they requested the first page if they did not include any pagination options. Its ok to break BC here.

**Comment by lyoshenka** [comment]
@osilkin98 do you know what to do next? do you want to talk to jack or lex?

**Comment by osilkin98** [comment]
> Feels like the solution is to pretend they requested the first page
Should the `page_size` be 10 or just the entire thing?

**Comment by osilkin98** [comment]
What are we thinking with this?

**Comment by osilkin98** [comment]
Because they were easier to regex than using brackets.

**Comment by eukreign** [comment]
@osilkin98 using direct dictionary access is better because it fails fast if the key is not there instead of returning None, also it's inconsistent with the rest of the code base which uses plain Python dictionary access except in specific circumstances where it's an expected case that the key might not be present.

can you please change the `.get()` to `[]`

**Comment by osilkin98** [comment]
Alright, give me a bit.

**Pull request merged**